src/packages/cli/src/args.ts

```diff
@@ -160,8 +165,9 @@ export default function (
 
   // disable dot-notation because yargs just can't coerce args properly...
   // ...on purpose! https://github.com/yargs/yargs/issues/1021#issuecomment-352324693
-  yargs
+  const yargsParser = yargs()
```

Comment:
Why shouldn't we use the singleton?

Reply:
Cos singletons are the literal devil 👿 

But actually, because the singleton makes it much harder to test. Because the yargs singleton underlying the args parser would be shared between each instance under test, we would need to reset it's state between tests (probably reaching into the yargs module directly from the test, which makes it more fragile, because the test is making assumptions about the implementation that aren't enforced). It's much nicer to just have each instance of the args parser to be self contained.